lib/core/minimal_config.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class MinimalConfig
```

Comment:
Why not edit `Config` instead? We could add a named param `minimal` (or whatever) to `initialize` (that defaults to `false`) and exit before `load_app_config` if it's `true`.

That way we have a single source of truth for config, minimal or not.

Reply:
I wasn't sure if that would be a good design.
Perhaps a class that could behave differently based on a flag should become two classes.
However, this implementation also is not ideal for me.

Parallelly, I am working on a different way of handling configuration. I think it is cleaner and more flexible. But I will propose it when I make the whole system work together.

Reply:
Ok, maybe a better implementation:

- Update `MinimalConfig` to be exactly like `Config` before `load_app_config`
- Have `Config` extend `MinimalConfig`

That way, we'd have the same source of truth for the minimal config whether or not `with_minimal_config` is `true`, and we'd still be separating the contexts.

> Parallelly, I am working on a different way of handling configuration.

Maybe it would be best for us to discuss it first, otherwise you could be spending time on something that might not be agreed upon. Plus, I did some changes to `Config` in #109.

Reply:
Yes. MinimalConfig is a kind of super class for config. Perhaps the name should get changed to `Config` and `ExtendedConfig`.

I have another idea for the configuration class. I will share it with you.